### ISSUE TITLE
Test failure on CI

### ISSUE THREAD

**Issue by lehins** [IntersectMBO/cardano-ledger#3910]
```
    DRep property traces
      Watch pulser on traces of length 120.:                               FAIL (0.68s)
        *** Failed! (after 1 test):
        Exception:
          Prelude.tail: empty list
          CallStack (from HasCallStack):
            error, called at libraries/base/GHC/List.hs:1644:3 in base:GHC.List
            errorEmptyList, called at libraries/base/GHC/List.hs:130:28 in base:GHC.List
            tail, called at src/Test/Cardano/Ledger/Constrained/Trace/TraceMonad.hs:447:63 in cardano-ledger-test-9.9.9.9-inplace:Test.Cardano.Ledger.Constrained.Trace.TraceMonad
        Use --quickcheck-replay=433821 to reproduce.
        Use -p '/Watch pulser on traces of length 120./' to rerun this test only.
```

**Comment by Soupstraw** [comment]
Another failure [here](https://github.com/input-output-hk/cardano-ledger/actions/runs/7113389250/job/19365233325?pr=3852#step:15:4830)

**Comment by lehins** [comment]
This test was fixed in #3912
